### DISCUSSION TITLE
Hash accelerometer

### DISCUSSION THREAD

**Discussion by danteppc** [bitcraze/discussions#152] [Q&A] [answered]
Hello,

It seems the STM32F405RG chip on Crazyflie is lacking the hardware for the hash functions such as MD5 or SHA1 [here](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/lib/STM32F4xx_StdPeriph_Driver/inc/stm32f4xx_hash.h) to work, hence they are never compiled.

Just wondering if there are plans to upgrade the chip on Crazyflie to something like STM32F415?

Also, would there happen to be a a software-based MD5 or SHA1 in the source code? I couldn't find.

Thank you

**Comment by krichardsson** [comment, marked as answer]
HI!

I don't think there is any implementation of MD5 or SHA1, but you can probably add one easily.

There are no immediate plans to update the MCU, but I suppose we will do it at some point, not sure when or to what model though.
